### PR TITLE
OSDOCS-11760: Admin/Dev console 4.17 release notes

### PR DESCRIPTION
**Pull request by opayne1**
Combines addition of Addmin and dev console release notes

Version(s):
4.17

Issue:
[Web console 4.17 release notes](https://issues.redhat.com/browse/OSDOCS-11760)

Link to docs preview:
https://82377--ocpdocs-pr.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-17-release-notes.html#ocp-4-17-web-console_release-notes

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by opayne1** [comment]
Hi @sanketpathak and @yapei could you help with a QE review of this PR? It has both admin and dev console notes in it. Thanks!

File: release_notes/ocp-4-17-release-notes.adoc [hunk 1]
```diff
@@ -101,16 +101,47 @@ With this release, the installation program uses Cluster API instead of Terrafor
 [id="ocp-4-17-web-console_{context}"]
 === Web console
 
+[id="ocp-4-17-ocp-lightspeed_{context}"]
+==== {ols} is available in the web console
+
+Starting with {product-title} 4.16, {ols} is available for use in the web console.
```
Comment (yapei) [inline release_notes/ocp-4-17-release-notes.adoc:107]: Shall we add some new features about OLS here or in `Administrator perspective` section? 
For example [PR](https://github.com/openshift/console/pull/14030) addes Lightspeed hover button and user preference settings to show/hide lightspeed button

Comment (yapei) [inline release_notes/ocp-4-17-release-notes.adoc:107]:
```suggestion
Starting with {product-title} 4.16, {ols} operator is available for use in the web console.
```

File: release_notes/ocp-4-17-release-notes.adoc [hunk 2]
```diff
@@ -101,16 +101,47 @@ With this release, the installation program uses Cluster API instead of Terrafor
...
+
+//https://github.com/openshift/openshift-docs/pull/81581
+//For more information, see xref:../web_console/capabilities_products-web-console.html#openshift-lightspeed-web-console_capabilities-web-console[Enhancing your experience in the {product-title} web console by using {ols-official}].
+
 [id="ocp-4-17-administrator-perspective_{context}"]
 ==== Administrator perspective
 
 This release introduces the following updates to the *Administrator* perspective of the web console:
 
+* Deprecated Operators are displayed in the OperatorHub before and after installation.
```
Comment (yapei) [inline release_notes/ocp-4-17-release-notes.adoc:117]:
```suggestion
* Deprecated Operators will be displayed along with a warning icon in the OperatorHub before and after installation.
```

Comment (yapei) [reply release_notes/ocp-4-17-release-notes.adoc:117]: We may need to clarify what's the difference between deprecated operators and normal operators(made a suggestion above)
Screenshot 2024-09-25 at 10 37 15 AM

File: release_notes/ocp-4-17-release-notes.adoc [hunk 3]
```diff
@@ -101,16 +101,47 @@ With this release, the installation program uses Cluster API instead of Terrafor
...
+* Config files for `MachineConfig` objects were moved to a new tab enabling you to check the content of the configuation files without having to manually retrieve its contents.
```
Comment (yapei) [inline release_notes/ocp-4-17-release-notes.adoc:118]:
```suggestion
* Config files for `MachineConfig` objects were moved to a new section enabling you to check the content of the configuation files without having to manually retrieve its contents.
```

Comment (yapei) [reply release_notes/ocp-4-17-release-notes.adoc:118]: Screenshot 2024-09-25 at 10 42 24 AM

File: release_notes/ocp-4-17-release-notes.adoc [hunk 4]
```diff
@@ -101,16 +101,47 @@ With this release, the installation program uses Cluster API instead of Terrafor
...
+* An alert was added to `operator-hub-item-details` and `operator-hub-subscribe` components if your cluster is on {gcp-first} with Workload Identity Foundation (WIF).
+* A page for ShipWright `BuildStrategy` was added with a `ClusterBuildStrategy` and `BuildStrategy` tab.
```
Comment (yapei) [inline release_notes/ocp-4-17-release-notes.adoc:120]: Is this Dev Console new change? admin console doesn't have this new feature in 4.17

File: release_notes/ocp-4-17-release-notes.adoc [hunk 5]
```diff
@@ -101,16 +101,47 @@ With this release, the installation program uses Cluster API instead of Terrafor
...
+//For more information, see xref:../web_console/capabilities_products-web-console.html#openshift-lightspeed-web-console_capabilities-web-console[Enhancing your experience in the {product-title} web console by using {ols-official}].
+
 [id="ocp-4-17-administrator-perspective_{context}"]
 ==== Administrator perspective
 
 This release introduces the following updates to the *Administrator* perspective of the web console:
 
+* Deprecated Operators are displayed in the OperatorHub before and after installation.
+* Config files for `MachineConfig` objects were moved to a new tab enabling you to check the content of the configuation files without having to manually retrieve its contents.
+* An alert was added to `operator-hub-item-details` and `operator-hub-subscribe` components if your cluster is on {gcp-first} with Workload Identity Foundation (WIF).
```
Comment (yapei) [inline release_notes/ocp-4-17-release-notes.adoc:119]: Shall we use pages instead of component names in release note?

Comment (yapei) [inline release_notes/ocp-4-17-release-notes.adoc:119]:
```suggestion
* An alert will be shown on Operator details and Operator installation page if your cluster is on {gcp-first} with Workload Identity Foundation (WIF).
```

Comment (yapei) [reply release_notes/ocp-4-17-release-notes.adoc:119]: **operator details modal**
Screenshot 2024-09-25 at 10 48 04 AM

**operator installation page**
Screenshot 2024-09-25 at 10 48 09 AM

File: release_notes/ocp-4-17-release-notes.adoc [hunk 6]
```diff
@@ -101,16 +101,47 @@ With this release, the installation program uses Cluster API instead of Terrafor
...
+* A page for ShipWright `BuildStrategy` was added with a `ClusterBuildStrategy` and `BuildStrategy` tab.
+
+[id="ocp-4-17-customize-create-project-modal-dyanmic-plugin"]
+===== Customize the *Create project* modal using dynamic plugins
+
+With this release, a new extension point was added, so dynamic plugin creators can pass a component that renders in place of the default *Create project* modal.
```
Comment (yapei) [inline release_notes/ocp-4-17-release-notes.adoc:125]:
```suggestion
With this release, a new extension point was added, so dynamic plugin creators can pass a component that renders in place of the default *Create Project* modal.
```
Screenshot 2024-09-25 at 10 50 11 AM

File: release_notes/ocp-4-17-release-notes.adoc (hunk 5 above)
Comment (opayne1) [reply release_notes/ocp-4-17-release-notes.adoc:119]: I see what you are saying! Okay, that makes sense! Thanks!

File: release_notes/ocp-4-17-release-notes.adoc (hunk 4 above)
Comment (opayne1) [reply release_notes/ocp-4-17-release-notes.adoc:120]: I see it was added with this PR https://github.com/openshift/console/pull/14029

@sanketpathak could you weigh in on this?

Comment (sanketpathak) [reply release_notes/ocp-4-17-release-notes.adoc:120]: Yes this feature is added as part of epic [ODC-7518](https://issues.redhat.com/browse/ODC-7518)

File: release_notes/ocp-4-17-release-notes.adoc [hunk 7]
```diff
@@ -101,16 +101,47 @@ With this release, the installation program uses Cluster API instead of Terrafor
...
+===== External OpenID Connect (OIDC) token issuer is now functional in the web console
+
+With this update, the web console works as expected when internal `oauth-server` and `oauth-apiserver` are removed and replaced with an external OpenID Connect (OIDC) issuer.
+
 [id="ocp-4-17-developer-perspective_{context}"]
 ==== Developer Perspective
 
 This release introduces the following updates to the *Developer* perspective of the web console:
 
+* When you use one of the import flows to create a new deployment, the *Import from Git* or *Container images* automatically opens them in the sidebar.
```
Comment (sanketpathak) [inline release_notes/ocp-4-17-release-notes.adoc:139]:
```suggestion
* When you use one of the add flows to create a new deployment, the *Import from Git* or *Container images* automatically opens them in the sidebar.
```

File: release_notes/ocp-4-17-release-notes.adoc [hunk 8]
```diff
@@ -101,16 +101,47 @@ With this release, the installation program uses Cluster API instead of Terrafor
...
+* Easily select the desired *Git Type* without having to use the list if {product-title} is unable to identify it.
+* *Import from Git* supports GitEA, an open-source alternitve to GitHub.
+* A warning notification displays on the *Topology* page if the `PodDisruptionBudget` limit is reached.
+* Use Shipwright Builds when importing applications through the *Import from Git* flows to use Shipwright Build strategies such as buildpacks when building your application.
```
Comment (sanketpathak) [inline release_notes/ocp-4-17-release-notes.adoc:143]:
```suggestion
* Use Shipwright Builds when importing applications through the *Import from Git* flows to use Shipwright Build strategies such as S2I, buildpack and Dockerfile (buildah) strategy for building the image```

File: release_notes/ocp-4-17-release-notes.adoc [hunk 9]
```diff
@@ -101,16 +101,47 @@ With this release, the installation program uses Cluster API instead of Terrafor
 [id="ocp-4-17-web-console_{context}"]
 === Web console
 
+[id="ocp-4-17-ocp-lightspeed_{context}"]
+==== {ols} Operator is available in the web console
+
+Starting with {product-title} 4.16, {ols} Operator is available for use in the web console. With this release, a hover button was added to help you discover {ols}. Once you click the hover button, the chat window will appear with instructions on how to enable and install {ols} on the cluster. You can hide the {ols} button when changing the default user preferences.
```
Comment (mburke5678) [inline release_notes/ocp-4-17-release-notes.adoc:107]: What is a hover button?

Comment (mburke5678) [reply release_notes/ocp-4-17-release-notes.adoc:107]:
```suggestion
Starting with {product-title} 4.16, {ols} Operator is available for use in the web console. With this release, a hover button was added to help you discover {ols}. Once you click the hover button, the chat window appears with instructions on how to enable and install {ols} on the cluster. You can hide the {ols} button when changing the default user preferences.
```

File: release_notes/ocp-4-17-release-notes.adoc [hunk 10]
```diff
@@ -101,16 +101,47 @@ With this release, the installation program uses Cluster API instead of Terrafor
...
+
+//https://github.com/openshift/openshift-docs/pull/81581
+//For more information, see xref:../web_console/capabilities_products-web-console.html#openshift-lightspeed-web-console_capabilities-web-console[Enhancing your experience in the {product-title} web console by using {ols-official}].
+
 [id="ocp-4-17-administrator-perspective_{context}"]
 ==== Administrator perspective
 
 This release introduces the following updates to the *Administrator* perspective of the web console:
 
+* Deprecated Operators are displayed in the OperatorHub before and after installation along with a warning notification the Operator is deprecated.
```
Comment (mburke5678) [inline release_notes/ocp-4-17-release-notes.adoc:117]:
```suggestion
* Deprecated Operators are displayed in the OperatorHub before and after installation along with a warning notification that the Operator is deprecated.
```

File: release_notes/ocp-4-17-release-notes.adoc [hunk 11]
```diff
@@ -101,16 +101,47 @@ With this release, the installation program uses Cluster API instead of Terrafor
...
+* Config files for `MachineConfig` objects were moved to a new section enabling you to check the content of the configuation files without having to manually retrieve its contents.
```
Comment (mburke5678) [inline release_notes/ocp-4-17-release-notes.adoc:118]: Where were they moved to?

File: release_notes/ocp-4-17-release-notes.adoc (hunk 4 above)
Comment (mburke5678) [reply release_notes/ocp-4-17-release-notes.adoc:120]: Will users know where to look for the new page?

File: release_notes/ocp-4-17-release-notes.adoc [hunk 12]
```diff
@@ -101,16 +101,47 @@ With this release, the installation program uses Cluster API instead of Terrafor
...
+===== Customize the *Create project* modal using dynamic plugins
+
+With this release, a new extension point was added, so dynamic plugin creators can pass a component that renders in place of the default *Create Project* modal.
+
+For more information on {product-title} Console dynamic plugin SDK extensions, see xref:../web_console/dynamic-plugin/dynamic-plugins-reference.adoc#dynamic-plugin-sdk-extensions_dynamic-plugins-reference[Dynamic plugin extension types].
+
+[id="ocp-4-17-console-functional-external-oidc-issuer"]
+===== External OpenID Connect (OIDC) token issuer is now functional in the web console
+
+With this update, the web console works as expected when internal `oauth-server` and `oauth-apiserver` are removed and replaced with an external OpenID Connect (OIDC) issuer.
```
Comment (mburke5678) [inline release_notes/ocp-4-17-release-notes.adoc:132]: `oauth-server` and `oauth-apiserver` Each needs a noun.

File: release_notes/ocp-4-17-release-notes.adoc [hunk 13]
```diff
@@ -101,16 +101,47 @@ With this release, the installation program uses Cluster API instead of Terrafor
...
+
 [id="ocp-4-17-developer-perspective_{context}"]
 ==== Developer Perspective
 
 This release introduces the following updates to the *Developer* perspective of the web console:
 
+* When you use one of the import flows to create a new deployment, the *Import from Git* or *Container images* automatically opens them in the sidebar.
+* Easily select the desired *Git Type* without having to use the list if {product-title} is unable to identify it.
```
Comment (mburke5678) [inline release_notes/ocp-4-17-release-notes.adoc:140]: Make this a complete sentence:
```suggestion
* You can easily select the desired *Git Type* without having to use the list if {product-title} is unable to identify it.
```

Comment (mburke5678) [reply release_notes/ocp-4-17-release-notes.adoc:140]: Also, ambiguous _it_. And will users know what a Git Type is and where to select it?

File: release_notes/ocp-4-17-release-notes.adoc (hunk 8 above)
Comment (mburke5678) [inline release_notes/ocp-4-17-release-notes.adoc:143]: Make this a complete sentence:
```suggestion
* You can use Shipwright Builds when importing applications through the *Import from Git* flows to use Shipwright Build strategies such as buildpacks when building your application.
```

Comment (mburke5678) [reply release_notes/ocp-4-17-release-notes.adoc:143]: Maybe split this into two sentences? 
```suggestion
* You can use Shipwright Builds when importing applications through the *Import from Git* flows. Also, you can use Shipwright Build strategies such as buildpacks when building your application.
```

**Comment by mburke5678** [comment]
@opayne1 A few comments. Otherwise LGTM.

**Pull request merged**